src/exportJarFileCommand.ts

```diff
@@ -32,33 +32,26 @@ export async function executeExportJarTask(node?: INodeData): Promise<void> {
     if (!isStandardServerReady() || isExportingJar || await buildWorkspace() === false) {
         return;
     }
+    isExportingJar = true;
     const stepMetadata: IStepMetadata = {
         entry: node,
         steps: [],
     };
-    tasks.executeTask(ExportJarTaskProvider.getTask(stepMetadata));
-    return;
-}
-
-export async function createJarFile(stepMetadata: IStepMetadata) {
-    isExportingJar = true;
-    let step: ExportJarStep = ExportJarStep.ResolveJavaProject;
-    return new Promise<string>(async (resolve, reject) => {
-        while (step !== ExportJarStep.Finish) {
-            try {
-                step = await stepMap.get(step).execute(stepMetadata);
-            } catch (err) {
-                return reject(err);
-            }
-        }
-        return resolve(stepMetadata.outputPath);
-    }).then((message) => {
-        isExportingJar = false;
-        successMessage(message);
-    }, (err) => {
-        isExportingJar = false;
+    try {
+        await stepMap.get(ExportJarStep.ResolveJavaProject).execute(stepMetadata);
+    } catch (err) {
         if (err) {
             failMessage(`${err}`);
         }
-    });
+        isExportingJar = false;
+        return;
+    }
+    tasks.executeTask(ExportJarTaskProvider.getTask(stepMetadata));
+}
+
+export async function finishExportJarTask(restart: boolean, node?: INodeData): Promise<void> {
+    isExportingJar = false;
+    if (restart) {
+        executeExportJarTask(node);
```

Comment:
Currently, the `node` here is always undefined because only the export jar process triggered by the button in the title of the Project Manager (not by the inline button of workspaceNode) could go back to the `ResolveJavaProject` step.
Should the `node` be `undefined` or just keep it for the consistency here?

Reply:
This function is a trouble. Some reasons:

- Still, it makes the people have to read the code between `exportJarFileCommand.ts` & `ExportJarTaskProvider.ts` back and forth
- It's named `finishExportJarTask` but with a param `restart`, confusing here.